### ISSUE TITLE
Fluffychat bugged after logging back in

### ISSUE THREAD

**Issue by gholmann16** [krille-chan/fluffychat#1012]
### Bug Description

After signing in, syncing my chats (thankfully everything was backed up), and shutting down my computer, when I logged in again fluffychat was no longer logged in. Of note: I had two users logged in at once. 

Error:
```
flutter: error SqfliteFfiException(sqlite_error: 26, , SqliteException(26): while selecting from statement, file is not a database, file is not a database (code 26)
  Causing statement: PRAGMA user_version}) DatabaseException(SqliteException(26): while selecting from statement, file is not a database, file is not a database (code 26)
  Causing statement: PRAGMA user_version) sql 'PRAGMA user_version' {details: {database: {path: /home/gholm/.local/share/chat.fluffy.fluffychat/FluffyChat linux.sqlite, id: 1, readOnly: false, singleInstance: true}}} during open, closing...
```

In the app, I tried to sign in but when I put in my credentials it would kick me out, and when I clicked log in for a second time, it said server was null.

### Steps to Reproduce

1. Download fluffychat-bin off of aur
2. Log in with 2 users
3. Back up chats on both users
4. Reboot and launch
5. You're logged out of your accounts
6. Ignore errors and try to log in again
7. Login fails, try again
8. Server is now "null" even if you specified custom server

### Expected Behavior

I expected to be able to log in. 

### App Version

1.19.0-3

### Additional Platform Information

Linux Laptop

### Additional Context

Errors reported in notifications:
![fluffychat](https://github.com/krille-chan/fluffychat/assets/88942093/7261c8e7-ca21-400e-b7ff-643ccb702b3b)

**Comment by jiteshsingh** [comment]
for me deleting the sqlite db only temporarily fixes the issue.
the issue re-occurs randomly after rebooting the device.

**Comment by gholmann16** [comment]
@jiteshsingh same here

**Comment by chovanecadam** [comment]
Indeed, the bug is still present. Is there any additional info we can provide?

**Comment by xeTaiz** [comment]
can confirm this is still an issue in 1.22.1-1 (AUR)
deleting the cache does not solve the problem, the sqlite gets corrupted after the next login + reboot cycle.
I'm curious if anyone found a workaround? with the flatpak also affected it seems to be a linux-wide issue?

**Comment by projectnowhub** [comment]
Faced this issue in my Android app while opening the application.

<img width="540" height="1200" alt="Image" src="https://github.com/user-attachments/assets/b9353579-0559-430a-b3f2-34dfad49f475" />

**Comment by balgerion** [comment]
Anyone have some working workaround for that ? :)

**Comment by seaofcolors** [comment]
> can confirm this is still an issue in 1.22.1-1 (AUR) deleting the cache does not solve the problem, the sqlite gets corrupted after the next login + reboot cycle. I'm curious if anyone found a workaround? with the flatpak also affected it seems to be a linux-wide issue?

I'm on Arch Linux as well. With fluffychat-bin 2.4.1-1 I have been running into the same issue:
After reboot, fluffychat isn't logged in anymore. After forwarding me to the matrix login page in the browser, fluffychat pretends to be on new device. All chats are encrypted and locked until the "new device" is trusted by all other clients and the chat partners come online. This makes the app basically unuseable.

**Comment by Meisterlala** [comment]
I get `error SqfliteFfiException(sqlite_error: 26, , SqliteException(26): while selecting from statement, file is not a database, file is not a database (code 26)` whenever i force quit the application, for example when shutting down my PC.

I'm on arch linux and this happens both with the AUR package as well as the flatpak. This causes all my messages to disappear and i need to log into the matrix again.

Edit: the problem was a faulty keyring. After solving that the issue went away

**Comment by rasos** [comment]
I had repeatedly a similar issue with a database lock when starting up:

> [ERROR:flutter/runtime/dart_vm_initializer.cc(40)] Unhandled Exception: SqfliteFfiException(sqlite_error: 5, , SqliteException(5): while selecting from statement, database is locked, database is locked (code 5)       
>  Causing statement: PRAGMA user_version}) DatabaseException(SqliteException(5): while selecting from statement, database is locked, database is locked (code 5)

The FluffyChat  window remained black. It turned out that an instance was already running but it seemed not having launched. So I killed it, and did a new login.

```
ps aux | grep fluffy
myuser         7585 20.8  1.8 2953200 292472 ?      SLl  08:15   1:39 /snap/fluffychat/2176/bin/fluffychat
kill 7585
```

I had to do a fresh login (at least it remembered my home server) and it took 20 minutes to re-build the database and to download and de-crypt the messages. I also had to re-verify.

**Comment by garymoon** [comment]
Hiding [Meisterlala's reply](https://github.com/krille-chan/fluffychat/issues/1012#issuecomment-4069112840) as resolved makes sense, but also obscures a potential cause.

> Edit: the problem was a faulty keyring. After solving that the issue went away

@Meisterlala if there's any more information you could share I'd be much obliged 🙏 I'm running plasma and reliably have this happen. To my knowledge my GNOME keychain is OK (can be accessed via Seahorse).

So far closing FluffyChat with `pkill` has allowed me to configure three accounts in between `pkill`s. Fluffy doesn't respond to ctrl+q for me, but I also don't see any _quit_ or _exit_ type functionality in the UI. Is there a recommended way to exit the entire application gracefully please?

I've also noticed that timestamped backups of the db are made each startup. Can these be leveraged in case of this issue? I've backed up the keychain entry also in case it goes walkabout.

**Comment by Meisterlala** [comment]
@garymoon 
as far as i know, Fufflychat uses the Freedesktop Secret Service API. You can check what is listening for that on your system with `busctl --user status org.freedesktop.secrets`. 
And it can be accessed with [keepsecret](https://github.com/KDE/keepsecret). There i can see an entry from "chat.fluffy.fluffychat/FlutterSecureStorage"

I used to have a corrupt gnome-keyring. It didn't start correctly and apps could not write secrets to it. So i switched to using `kwallet6` with the `ksecretd` for the secret service compatibility. 
i disable the gnome-keyring by masking its service `systemctl --user mask gnome-keyring-daemon.service gnome-keyring-daemon.socket`

After that I needed to restart and login to Fluffychat again and all my problems went away.

**Comment by garymoon** [comment]
Wow, thank you @Meisterlala!! 💙

I was completely confident that kwallet was my secret provider, and I was very much mistaken 😅 I'm surprised Plasma doesn't warn you when you select kwallet as your secret provider but Gnome is already in there 😭 I ended up resorting to uninstalling gnome-keyring after nothing else seemed to stop it, and kwallet is now my secret provider. Saving the secret from Gnome and transferring it to kwallet with matching data dir worked, so I'll see how things go from here 🪅
